File: LC_217_Contains Duplicate.cpp
#include <vector>
// #include <unordered_set>
#include <unordered_map>


// store elements that aren't in the set in the set, if element is alr in the sem, then it's being repeated, hence return True, if no element is alr in the set, it passes, hence return false


class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // unordered_set<int> s;
        unordered_map<int,int> d;
        for (int i=0;i<nums.size();i++){
            if (d.count(nums[i])){
                return true;
            }
            d[nums[i]] = 1;
        }
        return false;
    }
};

File: Week_1/LC_217_Contains Duplicate.py
# https://leetcode.com/problems/contains-duplicate/solutions/3672475/4-methods-c-java-python-beginner-friendl-zozw
from typing import List

# store elements that aren't in the set in the set, if element is alr in the sem, then it's being repeated, hence return True, if no element is alr in the set, it passes, hence return false

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))

"""
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        di = {}
        for i in nums:
            if i in di:
                return True
            di[i] = 1
        return False
            
"""